pubsub/google/cloud/pubsub_v1/publisher/client.py

```diff
@@ -167,44 +214,62 @@ def target(self):
         """
         return publisher_client.PublisherClient.SERVICE_ADDRESS
 
-    def _batch(self, topic, create=False, autocommit=True):
-        """Return the current batch for the provided topic.
+    def _delete_sequencer(self, topic, ordering_key):
+        """ Called when a sequencer is done publishing all messages.
+            Removes the sequencer for the corresponding topic & ordering_key.
+        """
+        with self._batch_lock:
+            sequencer_key = (topic, ordering_key)
+            del self._sequencers[sequencer_key]
+
+    def _get_or_create_sequencer(self, topic, ordering_key):
+        """ Gets an existing sequencer or creates a new one given the (topic,
+            ordering_key) pair.
+        """
+        sequencer_key = (topic, ordering_key)
+        sequencer = self._sequencers.get(sequencer_key)
+        if sequencer is None:
+            if ordering_key == "":
+                # TODO: Consider implementing _delete_sequencer for unordered
+                # sequencer.
```

Comment:
Is this TODO something essential that was planned for this PR, or is it more a reminder for the future?

Reply:
It's for the future. I didn't want to add this feature because I'm trying to keep the old code as identical as possible. The old code did not have clean-up for the state per topic.